### ISSUE TITLE
Auth with custom claims?

### ISSUE THREAD

**Issue by afk-mario** [CSFrequency/react-firebase-hooks#33]
Is there any way to wait until the custom claims are loaded using ?

`user.getIdTokenResult()`

**Comment by nmaves** [comment]
Here is how I am pulling custom claims.

```javascript
  const [firebaseUser, loadingFirebaseUser] = useAuthState(auth)


  useEffect(() => {
    const fetchClaims = async () => {
      const {
        claims: { admin },
      } = await firebaseUser.getIdTokenResult()

      setIsAdmin(admin === true)
    }

    if (firebaseUser) fetchClaims()
  }, [db, firebaseUser])
```

**Comment by afk-mario** [comment]
Not 100% sure it works for everyone but this is my current hook

```js
export const useUserWithRole = (auth) => {
  const [user, loadingUser, errorUser] = useAuthState(auth);
  const [userWithRole, setUserWithRole] = useState();
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState();

  useEffect(() => {
    const getRole = async () => {
      if (errorUser) {
        setUserWithRole(undefined);
        setLoading(false);
      } else if (!user || loadingUser) {
        setUserWithRole(undefined);
        setLoading(true);
      } else {
        setLoading(true);
        try {
          const idTokenResult = await user.getIdTokenResult();
          setUserWithRole({ ...user, role: idTokenResult.claims.role });
          setLoading(false);
          setError(undefined);
        } catch (e) {
          setError(e);
          setLoading(false);
        }
      }
    };
    getRole();
  }, [user, auth, loadingUser, errorUser]);

  return [{ ...userWithRole }, loading, error];
};

```

**Comment by dohomi** [comment]
Hi Guys,

I sit in the same boat like you guys. I use Hasura with Firebase auth and looked for a possibility to process custom claim. I added a PR maybe you can check it out if that works for you. Here a small example in TypeScript
```ts
  const [firebaseUser, loadingFirebaseUser] = useAuthState(auth, {
    onUserChanged: async (user:firebase.User) => {
       // your custom claim logic
    }
  })

```

**Comment by chrisbianca** [comment]
Belatedly, I have just merged the #139 PR from @dohomi and this will be included in the upcoming v5 release in the next week or two. Apologies for the delay.

**Comment by chrisbianca** [comment]
v5 is now released: https://github.com/CSFrequency/react-firebase-hooks/releases/tag/v5.0.0
